src/vos/vos_aggregate.c

```diff
@@ -511,11 +586,22 @@ prepare_segments(struct agg_merge_window *mw)
 			ent_in->ei_inob = 0;
 		}
 
+		if (mw->mw_csum_support && ent_in->ei_inob != 0)
+			/* Allocates csum buf range for truncated segments. */
+			cs_total += csum_prepare_ent(ent_in, phy_ent);
+
 		io->ic_seg_cnt++;
 		D_ASSERT(io->ic_seg_cnt <= io->ic_seg_max);
 	}
+	if (mw->mw_csum_support && cs_total) {
```

Comment:
Yes, since it's stay zero unless we are supporting checksums.  However, the code is changed so the calls to csum_prepare_ent are within an if (mw->mw_csum_support) block. (Before, I was calling this inside the build segment loop -- too many times).  It's this call (within a loop) that calculates cs_total.  the call to prepare_buf is now called within if (cs_total).